internal/standalone/standalone.go

```diff
@@ -17,113 +18,153 @@ import (
+
+	if *standaloneEnabled {
+		s, err := serverSettings(pluginID)
 		if err != nil {
-			return info, err
+			log.Printf("Error: %s", err.Error())
+			return ServerSettings{}, false
 		}
-		executable = js
+		return s, true
 	}
-	info.Dir = filepath.Dir(executable)
+	return ServerSettings{}, false
+}
 
-	// Determine standalone address + PID
-	var err error
-	info.Address, err = getStandaloneAddress(id, info)
+// ClientModeEnabled returns standalone server connection settings if a standalone server is running and can be reached.
+func ClientModeEnabled(pluginID string) (ClientSettings, bool) {
+	s, err := clientSettings(pluginID)
 	if err != nil {
-		return info, err
+		return ClientSettings{}, false
+	}
+
+	return s, true
+}
+
+// RunDummyPluginLocator runs the standalone server locator that Grafana uses to connect to the standalone GRPC server.
+func RunDummyPluginLocator(address string) {
+	fmt.Printf("1|2|tcp|%s|grpc\n", address)
+	t := time.NewTicker(time.Second * 10)
+
+	for ts := range t.C {
+		fmt.Printf("[%s] using address: %s\n", ts.Format("2006-01-02 15:04:05"), address)
 	}
-	info.PID, err = getStandalonePID(info)
+}
+
+func serverSettings(pluginID string) (ServerSettings, error) {
+	curProcPath, err := currentProcPath()
+	if err != nil {
+		return ServerSettings{}, err
+	}
+
+	dir := filepath.Dir(curProcPath) // Default to current directory
+	if pluginDir, found := findPluginRootDir(curProcPath); found {
+		dir = pluginDir
+	}
+
+	address, err := createStandaloneAddress(pluginID)
 	if err != nil {
-		return info, err
+		return ServerSettings{}, err
 	}
-	return info, nil
+	return ServerSettings{
+		Address: address,
+		Dir:     dir,
+	}, nil
 }
 
-func GetInfo(id string) (Args, error) {
-	var standalone, debug bool
-	flag.BoolVar(&standalone, "standalone", false, "should this run standalone")
-	flag.BoolVar(&debug, "debug", false, "run in debug mode")
-	flag.Parse()
+// clientSettings will attempt to find a standalone server's address and PID.
+func clientSettings(pluginID string) (ClientSettings, error) {
+	procPath, err := currentProcPath()
+	if err != nil {
+		return ClientSettings{}, err
+	}
 
-	// standalone path
-	ex, err := os.Executable()
+	dir := filepath.Dir(procPath)
+
+	// Determine running standalone address + PID
+	standaloneAddress, err := getStandaloneAddress(pluginID, dir)
 	if err != nil {
-		return Args{}, err
+		return ClientSettings{}, err
 	}
 
-	return getInfo(id, ex, standalone, debug)
+	if standaloneAddress == "" {
+		return ClientSettings{}, fmt.Errorf("address is empty")
+	}
+
+	standalonePID, err := getStandalonePID(dir)
+	if err != nil {
+		return ClientSettings{}, err
+	}
+
+	return ClientSettings{
+		TargetAddress: standaloneAddress,
+		TargetPID:     standalonePID,
+	}, nil
 }
 
-// will check a few options to find the dist plugin json file
-func findPluginJSON(exe string) (string, error) {
+func currentProcPath() (string, error) {
```

Comment:
Nit: Is this really necessary? it's basically just calling `os.Executable` and nothing else

Reply:
Ah I just have it because it's re-used between both client and server